File: leetcode/LinkedListTests/LinkedList_83.cs
namespace LinkedListTests;

public class LinkedList_83
{
    class Solution {
        public ListNode? DeleteDuplicates(ListNode? head) {
            if(head is null) return head;

            var slow = head;
            var fast = head.next;
            while(fast is not null) {
                if(fast.val != slow.val) {
                    slow.next = fast;
                    slow = slow.next;
                }
                fast = fast.next;
            }
            slow.next = null;
            return head;
        }
    }
}

File: leetcodePython/LinkedList/linkedlist_83.py
from typing import Optional


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head

        prev = head
        current = head.next

        while current:
            if prev.val == current.val:
                prev.next = current.next
            else:
                prev = current
            current = current.next
        return head
